Add disappearing messages feature with UI and state management

adianasct opened the pull request:
- Introduced a new DisappearingMessagesPopup component for setting message retention periods.
- Added functionality to handle disappearing message settings in the Channel and Actions components.
- Implemented a formatDisappearingMessageTime helper function for user-friendly display of retention periods.
- Updated Redux actions and sagas to manage message retention state and send system messages when settings are changed.
- Included new SVG assets for badges and icons related to disappearing messages.

adianasct committed:
Add disappearing messages feature with UI and state management

- Introduced a new DisappearingMessagesPopup component for setting message retention periods.
- Added functionality to handle disappearing message settings in the Channel and Actions components.
- Implemented a formatDisappearingMessageTime helper function for user-friendly display of retention periods.
- Updated Redux actions and sagas to manage message retention state and send system messages when settings are changed.
- Included new SVG assets for badges and icons related to disappearing messages.

karensct committed:
Merge branch 'feature/poll-messages' of https://github.com/sceyt/sceyt-chat-react-uikit into feat/disappearing-messages

adianasct committed:
Refactor CustomRadio component to support checkbox style

- Introduced a new isCheckboxStyle variable to determine the styling based on borderRadius.
- Updated the CustomLabel component to conditionally render the TickIcon based on the new checkbox style.
- Cleaned up background color handling and ensured proper dimensions for SVG icons within the checkbox style.

adianasct committed:
Merge branch 'feat/disappearing-messages' of https://github.com/sceyt/sceyt-chat-react-uikit into feat/disappearing-messages

adianasct committed:
Update disappearing messages feature to include a 1-month timer option and improve UI feedback for message settings. Adjusted related components to reflect changes in timer options and display status of disappearing messages.

karensct commented on src/common/customRadio/index.tsx:
```diff
@@ -79,8 +85,26 @@ const CustomLabel = styled.label<{
   `}
+
+  ${(props) => {
+    const isCheckboxStyle = props.borderRadius && props.borderRadius !== '50%'
```
send parametr to CustomRadio component and related it change ui, do not need to do this checking

karensct commented on src/common/popups/disappearingMessages/index.tsx:
```diff
@@ -0,0 +1,301 @@
+        maxWidth='522px'
+        minWidth='522px'
+        width={selectedOption === 'custom' ? '522px' : undefined}
+        height={selectedOption === 'custom' ? '491px' : undefined}
```
why do you chnage popup sizes related custom option?

karensct commented on src/common/popups/disappearingMessages/index.tsx:
```diff
@@ -0,0 +1,301 @@
+              Auto-delete after
+            </Label>
+
+            {(['off', '1day', '1week', '1month'] as TimerOption[]).map((option) => (
```
add in some constant variable

karensct commented on src/components/Channel/index.tsx:
```diff
@@ -571,6 +578,11 @@ const Channel: React.FC<IChannelProps> = ({
           />
+          {!!channel?.messageRetentionPeriod && (
+            <DisappearingMessagesBadge>
+              <BadgeIcon color={accentColor} size='16px' />
```
do you use size='16px' here?

karensct commented on src/components/ChannelDetails/Actions/index.tsx:
```diff
@@ -484,6 +501,31 @@ const Actions = ({
             <React.Fragment>{staredMessagesIcon || <DefaultStarIcon />} Starred messages </React.Fragment>
           </ActionItem>
         )}
+        {!channel.isMockChannel &&
```
do you know what is isMockChannel?

karensct commented on src/helpers/index.tsx:
```diff
@@ -666,3 +666,79 @@ export const hashString = async (str: string) => {
+  }
+
+  // Custom options (in seconds)
+  const CUSTOM_OPTIONS = [
```
everywhere you have added this constant values, please make it in constants ant export import from there

karensct commented on src/helpers/index.tsx:
```diff
@@ -666,3 +666,79 @@ export const hashString = async (str: string) => {
+  const periodInSeconds = periodInMilliseconds / 1000
+
+  // Fixed timer options (in seconds)
+  const FIXED_TIMER_OPTIONS: Record<string, number> = {
```
see comment below

adianasct committed:
Enhance message retention period handling in channel saga

- Added logic to normalize current and new message retention periods before setting them.
- Updated the saga to only call the setMessageRetentionPeriod method if the new period differs from the current one.
- Ensured that the channel data and all channels are updated accordingly when the retention period is set.

adianasct committed:
Enhance UI components for disappearing messages and custom radio options

- Added lineHeight prop to PopupName for better text alignment.
- Updated PopupBody to accept marginBottom prop for flexible spacing.
- Refactored CustomRadio to support variant prop for radio and checkbox styles.
- Improved DisappearingMessagesPopup to utilize new timer options structure and enhance UI feedback.
- Consolidated timer options into constants for better maintainability.

adianasct committed:
Update timer option labels in DisappearingMessagesPopup for improved readability

adianasct committed:
Refactor timer options in DisappearingMessagesPopup to use a constant array for improved maintainability and readability

karensct committed:
Merge branch 'feature/poll-messages' of https://github.com/sceyt/sceyt-chat-react-uikit into feat/disappearing-messages

karensct committed:
Update badge SVG dimensions and refactor DisappearingMessagesBadge component

- Changed badge SVG dimensions from 24x24 to 20x20 for better alignment.
- Refactored DisappearingMessagesBadge to accept color and strokeColor as props, enhancing customization.
- Adjusted positioning of the badge within the component for improved visual consistency.

adianasct committed:
Refactor DisappearingMessagesPopup to improve timer value handling

- Introduced useMemo hooks to calculate selected timer value and check if the value is unchanged.
- Updated the Set button's disabled state to reflect whether the timer value has changed, enhancing user interaction.

adianasct committed:
refactor

adianasct committed:
.

adianasct committed:
Refactor message retention period handling in saga

- Updated the message creation logic to directly use the provided periodInMilliseconds for autoDeletePeriod, improving clarity and consistency in message metadata.
- Removed redundant variable assignment for autoDeletePeriod, streamlining the code.

adianasct committed:
Update message body identifiers from 'DM' to 'ATM' for clarity in message retention logic

- Changed message body identifier in both Channel and SystemMessage components to 'ATM' to reflect the new terminology for auto-delete messages.
- Adjusted metadata handling in saga to ensure consistency with the updated message body identifier.

karensct committed:
Merge branch 'feature/poll-messages' of https://github.com/sceyt/sceyt-chat-react-uikit into feat/disappearing-messages

karensct committed:
Fix auto-delete message handling in SystemMessage component

- Updated the condition for displaying disappearing message status to correctly evaluate the autoDeletePeriod.
- Added a console log for debugging the autoDeletePeriod value.

karensct committed:
Update message body identifier from 'ATM' to 'ADM' for auto-delete messages

- Changed the message body identifier in Channel and SystemMessage components to 'ADM' to align with the updated terminology for auto-delete messages.
- Adjusted the message creation logic in the saga to reflect this change, ensuring consistency across the application.

adianasct committed:
Enhance DisappearingMessagesPopup timer handling with refs

- Added useRef hooks to track previous timer value and initialization state, improving the logic for handling timer changes.
- Updated useEffect hooks to ensure proper state management during initial renders and timer updates, enhancing the component's responsiveness.

adianasct committed:
Refactor DisappearingMessagesBadge and message handling in Channel component

- Simplified the DisappearingMessagesBadge component by removing the unused strokeColor prop.
- Updated conditions for displaying last message status to exclude system messages, improving clarity in message handling.
- Enhanced readability of the Channel component by streamlining conditional rendering logic.

adianasct committed:
Update SVG assets and enhance DisappearingMessagesBadge styling

- Modified badge.svg to include a linear gradient and updated the stroke properties for better visual consistency.
- Adjusted watch.svg to use currentColor for fill and stroke, improving adaptability to theme changes.
- Enhanced DisappearingMessagesBadge component in Channel to accept strokeColor and isLightMode props for dynamic styling based on background color.
- Updated DefaultWatchIcon to adjust size and color based on light mode, ensuring better visibility in different themes.

karensct committed:
Update badge SVG and enhance DisappearingMessagesBadge styling in Channel component

- Modified badge.svg to increase dimensions and adjust stroke properties for improved visibility.
- Updated DisappearingMessagesBadge to use color prop instead of strokeColor, enhancing styling flexibility.
- Adjusted badge positioning and size within the Channel component for better alignment with design specifications.
